e2e/specs/forward_msg_cache.spec.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+describe("ForwardMsgCache test", () => {
+  beforeEach(() => {
+    cy.loadApp("http://localhost:3000/");
+  });
+
+  it("can fetch missing cached messages from the server", () => {
+    cy.window().then((win) => {
+      win.streamlitDebug.clearForwardMsgCache();
+    });
+
+    cy.rerunScript();
+
+    cy.get("[data-testid='stStatusWidget']").should("contain", "Running...");
```

Comment:
Just for understanding, why is this expected to contain running and then suddenly it's not there? 

Is it because we rerun and then the running in the top right disappears?

Reply:
Yeah it's to ensure that the rerun actually completes. It turns out that we have a custom `cy.waitForRerun` command that does exactly this that I forgot about, so switched to using that instead